lib/galaxy/workflow/scheduling_manager.py

```diff
@@ -332,6 +331,12 @@ def __attempt_schedule(self, invocation_id, workflow_scheduler):
             workflow_invocation = session.get(model.WorkflowInvocation, invocation_id)
 
             try:
+                if workflow_invocation.state == workflow_invocation.states.CANCELLING:
+                    workflow_invocation.cancel_invocation_steps()
+                    workflow_invocation.state = workflow_invocation.states.CANCELLED
```

Comment:
```suggestion
                    workflow_invocation.mark_cancelled()
```

Otherwise `mark_cancelled()` seems unused.